### PR TITLE
chromium: update to 96.0.4664.45

### PR DESCRIPTION
**Pull request by msisov**
Release notes -
https://chromereleases.googleblog.com/2021/11/stable-channel-update-for-desktop.html

License changes:
---------------
Added licenses:
  * ios/third_party/native_closure_compiler/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/3173692

  * third_party/android_deps/libs/org_checkerframework_checker_qual/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/2945894

  * third_party/cast_core/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/3161412

  * third_party/jinja2/LICENSE.rst
  - https://chromium-review.googlesource.com/c/chromium/src/+/320516

Removed licenses:
  * third_party/jinja2/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/3205163

Updated licenses:
  * third_party/tflite_support/LICENSE
  - https://chromium-review.googlesource.com/c/chromium/src/+/3152615
---------------

Test-built with:
* chromium-ozone-wayland:
  * honister,  clang, MACHINE="intel-corei7-64"
  * honister,  clang, MACHINE="raspberrypi3
  * hardknott, clang, MACHINE="raspberrypi3
  * dunfell,   clang, MACHINE="qemuarm64"
  * dunfell,   clang, MACHINE="m3ulcb"

* chromium-x11
  * honister,  clang, MACHINE="raspberrypi3"
  * hardknott, clang, MACHINE="raspberrypi3"
  * dunfell,   clang, MACHINE="raspberrypi3"

### CODE REVIEW

File: meta-chromium/recipes-browser/chromium/chromium-gn.inc [hunk 1]
```diff
@@ -21,8 +21,7 @@ SRC_URI += " \
     file://0011-exception_handler.cc-Match-the-types-for-SIGSTKSZ.patch \
     file://0013-Revert-Use-ffile-compilation-dir-instead-of-fdebug-c.patch \
     file://0014-ozone-wayland-don-t-build-xcb-for-pure-wayland-build.patch \
-    file://0016-Make-the-build-linux-unbundle-scripts-run-with-Pytho.patch \
-    file://jinja-python-3.10.patch \
+    file://0015-IWYU-add-memory-for-std-unique_ptr-in-base-CommandLi.patch \
```
Comment (rakuco) [inline meta-chromium/recipes-browser/chromium/chromium-gn.inc:24]: Is this one really needed? It looks like the sort of libstdc++ fix that wouldn't be necessary anymore.

Comment (msisov) [reply meta-chromium/recipes-browser/chromium/chromium-gn.inc:24]: it's needed for dunfell

File: meta-chromium/recipes-browser/chromium/files/chromium-wayland/0001-ozone-add-va-api-support-to-wayland.patch [hunk 2]
```diff
@@ -18,59 +18,31 @@ by Wayland later.
...
 diff --git a/media/gpu/vaapi/vaapi_picture_native_pixmap.cc b/media/gpu/vaapi/vaapi_picture_native_pixmap.cc
-index 941f24cc59590..42975746b5fb8 100644
+index 941f24cc59590..296db0d0ae77f 100644
 --- a/media/gpu/vaapi/vaapi_picture_native_pixmap.cc
 +++ b/media/gpu/vaapi/vaapi_picture_native_pixmap.cc
-@@ -12,6 +12,7 @@
- #include "ui/gfx/native_pixmap.h"
- #include "ui/gl/gl_bindings.h"
- #include "ui/gl/gl_image_native_pixmap.h"
-+#include "media/gpu/macros.h"
```
Comment (r1mikey) [inline meta-chromium/recipes-browser/chromium/files/chromium-wayland/0001-ozone-add-va-api-support-to-wayland.patch:69]: This is still needed, or the va-api build fails with the following:
```
../../media/gpu/vaapi/vaapi_picture_native_pixmap.cc:44:5: error: use of undeclared identifier 'VLOGF'
    VLOGF(1) << "Cannot sync VPP input surface";
    ^
../../media/gpu/vaapi/vaapi_picture_native_pixmap.cc:48:5: error: use of undeclared identifier 'VLOGF'
    VLOGF(1) << "Cannot convert decoded image into output buffer";
    ^
../../media/gpu/vaapi/vaapi_picture_native_pixmap.cc:54:5: error: use of undeclared identifier 'VLOGF'
    VLOGF(1) << "Cannot sync VPP output surface";
    ^
3 errors generated.
```

**Review by r1mikey — CHANGES_REQUESTED** [review]
It looks the the va-api churn breaks the build due to a missing `media/gpu/macros.h` include in `vaapi_picture_native_pixmap.cc`.

File: meta-chromium/recipes-browser/chromium/files/chromium-wayland/0001-ozone-add-va-api-support-to-wayland.patch (hunk 2 above)
Comment (msisov) [reply meta-chromium/recipes-browser/chromium/files/chromium-wayland/0001-ozone-add-va-api-support-to-wayland.patch:69]: Ah. I forgot to add use_vaapi gn arg to my build. Let me fix that. Sorry.

File: meta-chromium/recipes-browser/chromium/chromium-gn.inc (hunk 1 above)
Comment (rakuco) [reply meta-chromium/recipes-browser/chromium/chromium-gn.inc:24]: Ack, I also reproduced the failure here without the patch.

**Comment by r1mikey** [comment]
With the above fix, the va-api bits look good at runtime - i915.

**Comment by r1mikey** [comment]
> With the above fix, the va-api bits look good at runtime - i915.

And on Honister, corei7-64.

**Comment by msisov** [comment]
thanks. the patch has been updated

**Review by rakuco — APPROVED** [review]
lgtm from my side, thanks @msisov and @r1mikey

**Review by r1mikey — APPROVED** [review]
LGTM

**Pull request merged**